### PR TITLE
updated IoT section - short summary of Greengrass

### PR DESCRIPTION
**Pull request by lynnlangit**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1711,6 +1711,17 @@ IoT
 
 ![How AWS IoT Works](https://d0.awsstatic.com/IoT/diagrams/awsiot-how-it-works_HowITWorks_1-26.png "How AWS IoT Works")
 
+### IoT Greengrass
```
Comment (ThanosBaskous) [inline README.md]: "Thing Shadows" are mentioned a couple of times but not explained - would it be possible to add a brief clause and/or a documentation link to their first mention (or a separate bullet) explaining them?

File: README.md [hunk 2]
```diff
@@ -1711,6 +1711,17 @@ IoT
...
+    *  AWS Greengrass core software - includes a message manager object, lambda runtime, local copy service for IoT Thing shadows, and a deployment agent to manage Greengrass group configuration. Greengrass uses Greengrass Groups as containers for selected IoT devices settings, subscriptions and associated Lambda functions.  In a Greengrass group a device is either a Greengrass core or an IoT device which will be connected that particular Greengrass core.
+    *  AWS Greengrass core SDK - enables Lambda functions to interact with the AWS Greengrass core on which they run in order to publish messages, interact with the local Thing Shadows service, or invoke other deployed Lambda functions. NOTE: The AWS Greengrass Core SDK only supports sending MQTT messages with QoS = 0.
```
Comment (ThanosBaskous) [inline README.md]: "NOTE: The AWS Greengrass Core SDK only supports sending MQTT messages with QoS = 0."

Should this be an IoT tip/gotcha? Or maybe a separte bullet? Open to discussing.

File: README.md [hunk 3]
```diff
@@ -1711,6 +1711,17 @@ IoT
 
 ![How AWS IoT Works](https://d0.awsstatic.com/IoT/diagrams/awsiot-how-it-works_HowITWorks_1-26.png "How AWS IoT Works")
 
+### IoT Greengrass
+
+*	**Greengrass** is a software platform that extends AWS IoT capabilities allowing AWS Lambda functions to be run directly on local devices.  It also enables IoT devices to be able to securely communicate on a local network without having to connect to the cloud. Greengrass includes a local pub/sub message manager that can buffer messages if connectivity is lost so that inbound and outbound messages to the cloud are preserved. Locally deployed Lambda functions can be triggered by local events, messages from the cloud, or other sources. Greengrass includes secure authentication and authorization of devices within the local netwrok and also between the local network and the AWS cloud. It also provides secure, over-the-air software updates of Lambda functions.
```
Comment (ThanosBaskous) [inline README.md]: Typo: netwrok -> network

File: README.md (hunk 1 above)
Comment (lynnlangit) [inline README.md]: Actually I have provided a link earlier in this section to IoT device shadows, I can add the synonym in this section for clarity

Comment (ThanosBaskous) [inline README.md]: Perfect, thank you @lynnlangit!

**Comment by lynnlangit** [comment]
made all requested updates to PR

**Comment by ThanosBaskous** [comment]
Thanks @lynnlangit! Given that this is a new (sub)section, I'd like to see a second pass from another editor before merging. @QuinnyPig? @jlevy?

Side note: I don't think we should let it sit too long either, so I'll plan on merging it (and doing a follow-up review for my nits) in the next day or so if we don't hear anything - getting new product information into the guide quickly should be a priority).

**Comment by jlevy** [comment]
Looks like a great start to me. Made a few small copy edits.

I'm good with it as is, though I'd also suggest thinking:
- Can we convey how significant or useful this service is? Who uses it?
- Are there more AWS blogs/links that are relevant to include?
- Are there any other non-AWS blogs/info we should include to give perspective?

**Comment by jlevy** [comment]
Thanks @lynnlangit !!

**Comment by lynnlangit** [comment]
It's a new service, but needed to complete the IoT offering, as edge (or local node) message management significantly enhances any public-cloud IoT service offering by accounting for disconnected devices.  Azure has announced (but not shipped) a similar extension to it's Azure IoT suite.  GCP hasn't shipped it's IoT services yet.    The only non-AWS blog coverage I can find is from tech journalists (not practitioners).  I wasn't able to find any other useful AWS resources either.

**Comment by jlevy** [comment]
OK, yeah, my impression too. I added a 🐥 to remind about this. :) I'm good, let's merge @ThanosBaskous ?

**Comment by ThanosBaskous** [comment]
Yes! Thank you again for adding this, @lynnlangit!

**Pull request merged**